helper/schema/provider.go

```diff
@@ -243,12 +243,9 @@ func (p *Provider) ValidateResource(
 //
 // This won't be called at all if no provider configuration is given.
 func (p *Provider) Configure(ctx context.Context, c *terraform.ResourceConfig) diag.Diagnostics {
-
-	var diags diag.Diagnostics
-
 	// No configuration
 	if p.ConfigureFunc == nil && p.ConfigureContextFunc == nil {
-		return diags
+		return nil
```

Comment:
I swapped some of these back, I think returning `nil` when possible is more clear that its not a potential error case. Preserving the go idioms if possible is still useful. When someone does have warnings, the top level `diags` makes sense, but otherwise, I think it may be more effort and compromises on clarity.